cost/terminate_policy/server_terminate.pt

```diff
@@ -0,0 +1,188 @@
+end
+
+###############################################################################
+# Authentication
+###############################################################################
+
+auth "auth_rs", type: "rightscale"
+
+###############################################################################
+# Resources
+###############################################################################
+
+resources "clouds", type: "rs_cm.clouds"
+
+resources "instances", type: "rs_cm.instances" do
+  iterate @clouds  
+  cloud_href href(iter_item)
+  tags 'instance:terminate=*'
+  filter do
+    state "operational"
+  end
+  view 'extended'
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+datasource "ds_clouds" do
+  iterate @clouds
+  field "href", href(iter_item)
+  field "name", val(iter_item, "display_name")
+end
+
+datasource "ds_instances" do
+  iterate @instances
+  field "resource_uid", val(iter_item,  "resource_uid")
+  field "name", val(iter_item, "name")
+  field "href", href(iter_item)
+  field "cloud_href", jmes_path(iter_item, "links[?rel=='cloud'].href | [0]")
+  field "instance_type_href", jmes_path(iter_item, "links[?rel=='instance_type'].href | [0]")
+  field "locked", jmes_path(iter_item,"locked")
+  field "state", jmes_path(iter_item,"state")
+  field "tags", jmes_path(iter_item,"tags[?contains(@, 'instance:terminate') == `true`] | [0]")
+end
+
+datasource "ds_instances_with_timestamp" do
+  run_script $js_instances_with_timestamp, $ds_instances
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_instances_with_timestamp", type: "javascript" do
+  parameters "ds_instances"
+  result "results"
+  code <<-EOF
+    var results = []
+    var date = new Date()
+    var iso_date = date
+    for ( i = 0; i < ds_instances.length; i++ ) {
+      var instance = ds_instances[i]
+      if ( !instance.locked ){
+        var instance_timestamp = instance.tags.split("=")[1];
+        var instance_terminate_date = new Date(instance_timestamp);
+        console.log(typeof(instance_terminate_date))
+        var instance_terminate_date_iso = instance_terminate_date
+        console.log(instance_terminate_date+"-"+iso_date)
+        var should_terminate = false
+        if (  instance_terminate_date <= iso_date ) {
+          console.log("true")
+          var should_terminate = true
+        }
+        
+        results.push({
+          resource_uid: instance.resource_uid,
+          name: instance.name,
+          href: instance.href
+          cloud_href: instance.cloud_href,
+          instance_type_href: instance.instance_type_href,
+          locked: instance.locked,
+          state: instance.state,
+          tags: instance.tags,
```

Comment:
do you really want to show all tags, or just the terminate tag value, like terminate date?

Reply:
I filter to just that tag in the data source.